Example code cannot compile

When trying to run the example code showing in documentation,

```haskell
{-# LANGUAGE QuasiQuotes, TypeApplications #-}
module Main where

import Python.Inline
import Python.Inline.QQ

main :: IO ()
main = withPython $ do
  let input = [1..10] :: [Int]
      square :: Int -> Py Int 
      square x = pure $ x * x
  result <- runPy $ fromPy' @[Int] =<< [pye| [ square_hs(x) for x in input_hs ] |]
  print result
```

This example code throws the following compile time error:

```
app/Main.hs:13:29: error: [GHC-39999]
    • No instance for ‘ToPy (Int -> Py Int)’
        arising from a use of ‘inline-python-0.1:Python.Internal.EvalQQ.basicBindInDict’
        (maybe you haven't applied a function to enough arguments?)
    • In the expression:
        inline-python-0.1:Python.Internal.EvalQQ.basicBindInDict
          "square_hs" square
      In the second argument of ‘mapM_’, namely
        ‘[inline-python-0.1:Python.Internal.EvalQQ.basicBindInDict
            "square_hs" square,
          inline-python-0.1:Python.Internal.EvalQQ.basicBindInDict
            "input_hs" input]’
      In a stmt of a 'do' block:
        mapM_
          ($ p_dict_a2Mb)
          [inline-python-0.1:Python.Internal.EvalQQ.basicBindInDict
             "square_hs" square,
           inline-python-0.1:Python.Internal.EvalQQ.basicBindInDict
             "input_hs" input]
   |
13 |     fromPy' @[Int] =<< [pye| [ square_hs(x) for x in input_hs ] |]
   |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

However, after reading the documentation about ToPy class, it seems that there is an instance `ToPy (a -> IO b)`, we only need to change `square :: Int -> Py Int` to `square :: Int -> IO Int`, which compiles and runs as expected.

I'm using inline-python version 0.1, downloaded via hackage.

Btw this is a great project!

--- a/src/Python/Inline/Literal.hs
+++ b/src/Python/Inline/Literal.hs
@@ -577,7 +577,6 @@ instance (ToPy b) => ToPy (IO b) where
     --
     [CU.exp| PyObject* { inline_py_callback_METH_NOARGS($(PyCFunction f_ptr)) } |]
 
-
 -- | Only accepts positional parameters
 instance (FromPy a, Show a, ToPy b) => ToPy (a -> IO b) where
   basicToPy f = Py $ do
@@ -599,6 +598,40 @@ instance (FromPy a1, FromPy a2, ToPy b) => ToPy (a1 -> a2 -> IO b) where
       progPy $ basicToPy =<< dropGIL (f a1 a2)
     --
     [CU.exp| PyObject* { inline_py_callback_METH_FASTCALL($(PyCFunctionFast f_ptr)) } |]
+
+
+-- | Converted to 0-ary function
+instance (ToPy b) => ToPy (Py b) where
+  basicToPy f = Py $ do
+    --
+    f_ptr <- wrapCFunction $ \_ _ -> pyCallback $ do
+      progPy $ basicToPy =<< f
+    --
+    [CU.exp| PyObject* { inline_py_callback_METH_NOARGS($(PyCFunction f_ptr)) } |]
+
+-- | Only accepts positional parameters
+instance (FromPy a, Show a, ToPy b) => ToPy (a -> Py b) where
+  basicToPy f = Py $ do
+    --
+    f_ptr <- wrapCFunction $ \_ p_a -> pyCallback $ do
+      a <- loadArg p_a 0 1
+      progPy $ basicToPy =<< f a
+    --
+    [CU.exp| PyObject* { inline_py_callback_METH_O($(PyCFunction f_ptr)) } |]
+
+-- | Only accepts positional parameters
+instance (FromPy a1, FromPy a2, ToPy b) => ToPy (a1 -> a2 -> Py b) where
+  basicToPy f = Py $ do
+    --
+    f_ptr <- wrapFastcall $ \_ p_arr n -> pyCallback $ do
+      when (n /= 2) $ abortM $ raiseBadNArgs 2 n
+      a1 <- loadArgFastcall p_arr 0 n
+      a2 <- loadArgFastcall p_arr 1 n
+      progPy $ basicToPy =<< f a1 a2
+    --
+    [CU.exp| PyObject* { inline_py_callback_METH_FASTCALL($(PyCFunctionFast f_ptr)) } |]
+
+
 
 ----------------------------------------------------------------
 -- Helpers

--- a/test/TST/Callbacks.hs
+++ b/test/TST/Callbacks.hs
@@ -54,48 +54,59 @@ tests = testGroup "Callbacks"
          except TypeError as e:
              pass
          |]
+    ----------------------------------------
+  , testCase "Py function(arity 0)" $ do
+      let fun = [pye| 0 |]
+      runPy [py_| assert fun_hs() == 0 |]
+  , testCase "Py function(arity=1)" $ runPy $ do
+      let double (n::Int) = [pye| n_hs * 2 |]
+      [py_| assert double_hs(3) == 6 |]
+  , testCase "Py function(arity=2)" $ runPy $ do
+      let foo (x::Int) (y::Int) = [pye| x_hs * y_hs |]
+      [py_| assert foo_hs(3, 100) == 300 |]
+    ----------------------------------------
   , testCase "Haskell exception in callback(arity=1)" $ runPy $ do
-      let foo :: Int -> IO Int
-          foo y = pure $ 10 `div` y
-      throwsPy [py_| foo_hs(0) |]
-  , testCase "Haskell exception in callback(arity=2)" $ runPy $ do
-      let foo :: Int -> Int -> IO Int
-          foo x y = pure $ x `div` y
-      throwsPy [py_| foo_hs(1, 0) |]
-    ----------------------------------------
-  , testCase "Call python in callback (arity=1)" $ runPy $ do
-      let foo :: Int -> IO Int
-          foo x = do Just x' <- runPy $ fromPy =<< [pye| 100 // x_hs |]
-                     pure x'
-      [py_|
-        assert foo_hs(5) == 20
-        |]
-  , testCase "Call python in callback (arity=2" $ runPy $ do
-      let foo :: Int -> Int -> IO Int
-          foo x y = do Just x' <- runPy $ fromPy =<< [pye| x_hs // y_hs |]
-                       pure x'
-      [py_|
-        assert foo_hs(100,5) == 20
-        |]
-    ----------------------------------------
-  , testCase "No leaks (arity=1)" $ runPy $ do
-      let foo :: Int -> IO Int
-          foo y = pure $ 10 * y
-      [py_|
-        import sys
-        x = 123456
-        old_refcount = sys.getrefcount(x)
-        foo_hs(x)
-        assert old_refcount == sys.getrefcount(x)
-        |]
-  , testCase "No leaks (arity=2)" $ runPy $ do
-      let foo :: Int -> Int -> IO Int
-          foo x y = pure $ x * y
-      [py_|
-        import sys
-        x = 123456
-        old_refcount = sys.getrefcount(x)
-        foo_hs(1,x)
-        assert old_refcount == sys.getrefcount(x)
-        |]
-  ]
+       let foo :: Int -> IO Int
+           foo y = pure $ 10 `div` y
+       throwsPy [py_| foo_hs(0) |]
+   , testCase "Haskell exception in callback(arity=2)" $ runPy $ do
+       let foo :: Int -> Int -> IO Int
+           foo x y = pure $ x `div` y
+       throwsPy [py_| foo_hs(1, 0) |]
+     ----------------------------------------
+   , testCase "Call python in callback (arity=1)" $ runPy $ do
+       let foo :: Int -> IO Int
+           foo x = do Just x' <- runPy $ fromPy =<< [pye| 100 // x_hs |]
+                      pure x'
+       [py_|
+         assert foo_hs(5) == 20
+         |]
+   , testCase "Call python in callback (arity=2" $ runPy $ do
+       let foo :: Int -> Int -> IO Int
+           foo x y = do Just x' <- runPy $ fromPy =<< [pye| x_hs // y_hs |]
+                        pure x'
+       [py_|
+         assert foo_hs(100,5) == 20
+         |]
+     ----------------------------------------
+   , testCase "No leaks (arity=1)" $ runPy $ do
+       let foo :: Int -> IO Int
+           foo y = pure $ 10 * y
+       [py_|
+         import sys
+         x = 123456
+         old_refcount = sys.getrefcount(x)
+         foo_hs(x)
+         assert old_refcount == sys.getrefcount(x)
+         |]
+   , testCase "No leaks (arity=2)" $ runPy $ do
+       let foo :: Int -> Int -> IO Int
+           foo x y = pure $ x * y
+       [py_|
+         import sys
+         x = 123456
+         old_refcount = sys.getrefcount(x)
+         foo_hs(1,x)
+         assert old_refcount == sys.getrefcount(x)
+         |]
+   ]

